core/domain/exp_services.py

```diff
@@ -312,10 +312,9 @@ def get_exploration_summaries_matching_ids(exp_ids):
     summary domain objects (or None if the corresponding summary does not
     exist).
     """
-    return [
-        (get_exploration_summary_from_model(model) if model else None)
-        for model in exp_models.ExpSummaryModel.get_multi(exp_ids)]
-
+    return map(
+        get_exploration_summary_from_model,
+        filter(None, exp_models.ExpSummaryModel.get_multi(exp_ids)))
```

Comment:
I'm kind of confused here -- doesn't this remove entries which are None from exp_ids, and then retrieve the corresponding exploration summaries? So you end up with a list whose length may not match the length of exp_ids. 

I think the correct fix is to filter the list after you call this function. (In https://github.com/oppia/oppia/blob/develop/core/controllers/dashboard.py#L159 filter the list after you call get_exploration_summaries_matching_ids)

This is so that everything remains consistent with how these functions are used in the rest of the backend, so these function definitions don't change.

As as aside, I'm surprised the backend tests passed, both before the fix and at this point. Our tests really should cover these cases :)